homeassistant/components/kostal_plenticore/select.py

```diff
@@ -0,0 +1,129 @@
+"""Platform for Kostal Plenticore select widgets."""
+from __future__ import annotations
+
+from abc import ABC
+from datetime import timedelta
+import logging
+
+from homeassistant.components.select import SelectEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import DEVICE_DEFAULT_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, SELECT_SETTINGS_DATA
+from .helper import Plenticore, SelectDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Add kostal plenticore Select widget."""
+    plenticore: Plenticore = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        PlenticoreDataSelect(
+            hass=hass,
+            plenticore=plenticore,
+            entry_id=entry.entry_id,
+            platform_name=entry.title,
+            device_class="kostal_plenticore__battery",
+            module_id=select.module_id,
+            data_id=select.data_id,
+            name=select.name,
+            current_option="None",
+            options=select.options,
+            is_on=select.is_on,
+            device_info=plenticore.device_info,
+            unique_id=f"{entry.entry_id}_{select.module_id}",
+        )
+        for select in SELECT_SETTINGS_DATA
+    )
+
+
+class PlenticoreDataSelect(CoordinatorEntity, SelectEntity, ABC):
+    """Representation of a Plenticore Switch."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
```

Comment:
Don't pass in `hass`. It will be set on the entity when the entity has been added to home assistant. This means you can access `hass` with `self.hass`, after the entity has been added to home assistant.

Pass in the coordinator instead.

Reply:
Fixed in https://github.com/home-assistant/core/pull/59998